charmcraft/utils.py

```diff
@@ -57,6 +70,8 @@
 
 PathOrString = Union[os.PathLike, str]
 
+PACKAGE_LINE_REGEX = re.compile(r"^([A-Za-z0-9_.-]+)([<>=!]=[0-9.]+)?")
```

Comment:
There are also `~=` and `===` operators
https://peps.python.org/pep-0440/#version-specifiers

Also space should be ignored?

Reply:
You're right. I've changed this to cover all those cases and included tests for them.